op-supervisor/supervisor/backend/backend.go

```diff
@@ -441,9 +439,105 @@ func (su *SupervisorBackend) checkAccess(acc types.Access, execAt types.Executin
 	if err != nil {
 		return eth.BlockID{}, err
 	}
+
 	return bl.ID(), nil
 }
 
+type rpcCheckResult struct {
+	block eth.BlockID
+	err   error
+}
+
+func (su *SupervisorBackend) verifyAccessWithRPCInParallel(ctx context.Context, executingDescriptor types.ExecutingDescriptor, acc types.Access, primaryAccessCheck func(types.Access) (eth.BlockID, error)) (eth.BlockID, error) {
+	rpcResultChan := make(chan rpcCheckResult, 1)
+	go func() {
+		msgBlockRPC, err := su.checkAccessWithRPC(ctx, acc)
+		rpcResultChan <- rpcCheckResult{block: msgBlockRPC, err: err}
+	}()
+
+	msgBlockDB, err := primaryAccessCheck(acc)
+	if err != nil {
+		su.logger.Debug("Access-list inclusion check failed", "err", err)
+		return eth.BlockID{}, err
+	}
+
+	rpcResult := <-rpcResultChan
+	if rpcResult.err != nil {
+		su.logger.Error("RPC access check failed", "err", rpcResult.err)
+		return eth.BlockID{}, rpcResult.err
+	}
+	if msgBlockDB != rpcResult.block {
+		su.logger.Error("RPC access check failed, DB access check result did not match rpc access check result", "db_block", msgBlockDB, "rpc_block", rpcResult.block)
+		return eth.BlockID{}, fmt.Errorf("RPC access check failed, DB access check result did not match rpc access check result")
+	}
+	return rpcResult.block, nil
+}
+
+func (su *SupervisorBackend) checkAccessWithRPC(ctx context.Context, acc types.Access) (eth.BlockID, error) {
+	rpcAccessVerifyTimeout := 5 * time.Second
+
+	src, ok := su.syncSources.Get(acc.ChainID)
```

Comment:
I would like to see much of is `checkAccessWithRPC` functionality be implemented *inside* the `syncSource` that we are getting here. The Sync Source is a good place to store the details of getting receipts and checking them. See how we construct a `ContainsQuery` and give it to the Database, we could do the same thing here, which would allow for the Sync Source to mirror the database interface. I also think this will make it more testable, as you'll be able to create a mock SyncSource for the backend, and can test the Sync Start "Contains" method with better isolation.